routes.go

```diff
@@ -36,7 +36,7 @@ func setupRoutes(e *echo.Echo) {
 		r.GET("/sources", SourceList, tenancyWithListMiddleware...)
 		r.GET("/sources/:id", SourceGet, middleware.Tenancy)
 		r.POST("/sources", SourceCreate, permissionMiddleware...)
-		r.PATCH("/sources/:id", SourceEdit, permissionMiddleware...)
+		r.PATCH("/sources/:id", SourceEdit, append(permissionMiddleware, middleware.Notifier)...)
```

Comment:
I think we could make a new `editMiddleware` slice that contains this. Since we're using the same middleware for all the PATCH methods it looks like.